### PR TITLE
Run integration tests on juju 3.1 and 2.9

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Depends on https://github.com/canonical/mysql-operator/pull/255

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -60,7 +60,16 @@ jobs:
       actions: write  # Needed to manage GitHub Actions cache
 
   integration-test:
-    name: Integration test charm
+    strategy:
+      fail-fast: false
+      matrix:
+        juju-agent:
+          - 3.1.5
+          - 2.9.43
```
Comment (paulomach) [inline .github/workflows/ci.yaml:68]: It should be safe also to bump to `2.9.44` or there's some known issue?

**Review by paulomach — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (taurus-forever) [reply .github/workflows/ci.yaml:68]: +1 to move 2.9.44

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -60,7 +60,16 @@ jobs:
       actions: write  # Needed to manage GitHub Actions cache
 
   integration-test:
-    name: Integration test charm
+    strategy:
+      fail-fast: false
```
Comment (taurus-forever) [inline .github/workflows/ci.yaml:64]: Is it to count success rate on self-hosted runners?

File: pyproject.toml [hunk 3]
```diff
@@ -55,12 +55,13 @@ coverage = {extras = ["toml"], version = "^7.2.7"}
 [tool.poetry.group.integration.dependencies]
 pytest = "^7.4.0"
 pytest-operator = "^0.28.0"
-juju = "^2.9.44.0"
+juju = "^3.2.0.1"
```
Comment (taurus-forever) [inline pyproject.toml:58]: JFMI, where do we indicate old libjuju for juju 2.9 tests?

**Review by taurus-forever — DISMISSED** [review]

**Review by paulomach — APPROVED** [review]

**Pull request closed without merging**